### PR TITLE
support sles12 for statelite, nfs and ramdisk

### LINKED ISSUE #1329: [FVT]sles12.1 statellite genimage failed on x86_64
Test ENV:
```
c910f04x27v02:~ # cat /etc/*release
NAME="SLES"
VERSION="12-SP1"
VERSION_ID="12.1"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP1"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp1"
SUSE Linux Enterprise Server 12 (x86_64)
VERSION = 12
PATCHLEVEL = 1
# This file is deprecated and will be removed in a future service pack or release.
# Please check /etc/os-release for details about this release.
c910f04x27v02:~ # lsxcatd -a
Version 2.12.1 (git commit 5203da6cf25849c214d14b824f7b985c032bbbba, built Thu Jun 16 09:30:35 EDT 2016)
This is a Management Node
dbengine=SQLite
```
Description: after execute "genimage sles12.1-x86_64-statelite-compute", there are outputs below and command failed.
```
Creating initrd: /boot/initrd-3.12.49-11-default
Executing: /usr/bin/dracut --logfile /var/log/YaST2/mkinitrd.log --force /boot/initrd-3.12.49-11-default 3.12.49-11-default
Turning off host-only mode: '/sys' is not mounted!
Turning off host-only mode: '/proc' is not mounted!
Turning off host-only mode: '/run' is not mounted!
Turning off host-only mode: '/dev' is not mounted!
dracut module 'network' will not be installed, because command '/usr/lib/wicked/bin/wickedd-dhcp4' could not be found!
dracut module 'network' will not be installed, because command '/usr/lib/wicked/bin/wickedd-dhcp6' could not be found!
dracut module 'url-lib' will not be installed, because command 'curl' could not be found!
dracut module 'plymouth' will not be installed, because command 'plymouthd' could not be found!
dracut module 'plymouth' will not be installed, because command 'plymouth' could not be found!
dracut module 'crypt' will not be installed, because command 'cryptsetup' could not be found!
dracut module 'dmraid' will not be installed, because command 'dmraid' could not be found!
dracut module 'lvm' will not be installed, because command 'lvm' could not be found!
dracut module 'multipath' will not be installed, because command 'multipath' could not be found!
dracut module 'crypt-gpg' will not be installed, because command 'gpg' could not be found!
dracut module 'cifs' will not be installed, because command 'mount.cifs' could not be found!
dracut module 'fcoe' will not be installed, because command 'dcbtool' could not be found!
dracut module 'fcoe' will not be installed, because command 'fipvlan' could not be found!
dracut module 'fcoe' will not be installed, because command 'lldpad' could not be found!
dracut module 'fcoe-uefi' will not be installed, because command 'dcbtool' could not be found!
dracut module 'fcoe-uefi' will not be installed, because command 'fipvlan' could not be found!
dracut module 'fcoe-uefi' will not be installed, because command 'lldpad' could not be found!
dracut module 'iscsi' will not be installed, because command 'iscsi-iname' could not be found!
dracut module 'iscsi' will not be installed, because command 'iscsiadm' could not be found!
dracut module 'iscsi' will not be installed, because command 'iscsid' could not be found!
dracut module 'nbd' will not be installed, because command 'nbd-client' could not be found!
95nfs: Could not find any command of 'rpcbind portmap'!
dracut module 'ssh-client' will not be installed, because command 'ssh' could not be found!
dracut module 'ssh-client' will not be installed, because command 'scp' could not be found!
Failed to read /proc/cmdline, ignoring: No such file or directory
Failed to read /proc/cmdline. Ignoring: No such file or directory
*** Including module: bash ***
*** Including module: modsign ***
*** Including module: i18n ***
*** Including module: convertfs ***
*** Including module: ifcfg ***
*** Including module: btrfs ***
*** Including module: kernel-modules ***
```

### PR DESCRIPTION
**Pull request by penguhyang**
support dracut for sles12
fix #1329

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/sles.pm [hunk 1]
```diff
@@ -1722,6 +1732,19 @@ erver, if so, stop it first and try again" ],
     }
 }
 
+# Check whether the dracut is supported by this os
```
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/sles.pm]: whether using dracut should be determined by whether "dracut" directory exists inside the "rootimg" directory, refer to the relevant code slice in "rh/genimage"
````
 678 if ((-d "$rootimg_dir/usr/share/dracut") or (-d "$rootimg_dir/usr/lib/dracut")) {
 679     $dracutmode = 1;
 680
 681     # get dracut version
 682     $dracutver = `rpm --root $rootimg_dir -qi dracut | grep Version | awk -F' ' '{print \$3}'`;
 683     chomp($dracutver);
 684     if ($dracutver =~ /^\d\d\d$/) {
 685         if ($dracutver >= "033") {
 686             $dracutdir = "dracut_033";
 687         } else {
 688             $dracutdir = "dracut";    # The default directory
 689         }
 690     }
 691     print "Enter the dracut mode. Dracut version: $dracutver. Dracut directory: $dracutdir.\n";
 692 }
````

Comment (penguhyang) [inline xCAT-server/lib/xcat/plugins/sles.pm]: fixed.

File: xCAT-server/lib/xcat/plugins/sles.pm [hunk 2]
```diff
@@ -1722,6 +1732,19 @@ erver, if so, stop it first and try again" ],
...
+sub using_dracut
+{
+    my $rootimgdir = shift;
+    chdir("$rootimgdir");
+    my $outputmsg = `dracut --list-modules`;
```
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/sles.pm]: should better be `chroot $rootimgdir dracut --list-modules`

Comment (penguhyang) [inline xCAT-server/lib/xcat/plugins/sles.pm]: fixed.

**Comment by immarvin** [comment]
agree to merge

**Comment by daniceexi** [comment]
👍

**Pull request merged**